src/search/Search.tsx

```diff
@@ -1,7 +1,57 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+import { useSearchParams } from 'react-router-dom';
+
+import Spacing from '@/components/Spacing/Spacing';
+
+import { ResultListItem } from './components/ResultListItem/ResultListItem';
+import { SearchBar } from './components/SearchBar/SearchBar';
+import { useGetSearch } from './hooks/useGetSearch';
+import { NoResult } from './pages/NoResult/NoResult';
+
 export const Search = () => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+
+  const { data, isLoading, fetchNextPage, hasNextPage } = useGetSearch({
+    query: query as string,
+  });
+
+  const observer = useRef<IntersectionObserver>();
+
+  const lastElementRef = useCallback(
+    (node: HTMLDivElement) => {
+      if (isLoading) return;
+
+      if (observer.current) observer.current.disconnect(); // 🔵
```

Comment:
이........ 파랑콩떡의 정체가 뭘 의미하는지 모르겠는데 슬쩍 삭제 가능일까요

```suggestion
      if (observer.current) observer.current.disconnect();
```